src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
That makes sense. Thanks! Can you please add the tests you wrote to the set of automated tests?

This is still a deviation from spec though. You're ignoring maxDataLength and instead relying on the globally defined MAX value. If you go by spec, you'd pass maxDataLength as a parameter to valueParser and use that instead of MAX in valueParser implementation. So the code would look something like this:

```JavaScript
                    if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                      return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
                        return valueParse(parser, valueMetaData, options, maxDataLength + 1, callback);
                      });
                    } else {
                      return valueParse(parser, valueMetaData, options, MAX, callback);
                    }
```

Can you see how this is done SqlClient? Perhaps, we could mimic that here.